catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -146,28 +215,49 @@ public Metacard transform(InputStream inputStream)
     @Override
     public Metacard transform(InputStream inputStream, String id)
             throws IOException, CatalogTransformerException {
+        return handleTransform(inputStream, id);
+    }
 
-        String xml = null;
-        XstreamPathValueTracker pathValueTracker = null;
-        xml = IOUtils.toString(inputStream);
+    private Metacard handleTransform(InputStream inputStream, String id)
+            throws CatalogTransformerException {
+        String xml;
+        XstreamPathValueTracker pathValueTracker;
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+            readInputStreamIntoByteArrayOutputStream(inputStream, baos);
```

Comment:
Changed to use `IOUtils`.  While there is no guarantee of the size of the input stream, GMD files are xml and aren't likely to be XXGB.